### PR TITLE
AP-9907 # allow morph-expressions to work in commonJS and es-modules

### PR DESCRIPTION
**Pull request by divporter**
## Requester Checklist

Please only check the items that you have actioned. Do not check items that are not applicable to your PR.

### Implementation

- [ ] Have you tested your implementation locally?
- [ ] If applicable, has an appropriate changelog entry been added? Do not include if you are fixing/changing something that is only in the current release.
- [ ] There are no warnings that have been suppressed unnecessarily
- [ ] Have automated tests been added, or have related ones been updated to cover the change?
- [ ] Have all OneBlink dependency updates been completed (eg apps / apps-react / types etc).
- [ ] Have you ensured this change does not add unwanted dependencies?
- [ ] If this PR contains a refactor, have relevant Jira testing tasks added?
- [ ] Changes that will knowingly make the feature/bug incomplete have been commented with TODO and a description of what needs to be done to finish the feature/bug
- [ ] Any changes made to public APIs have been reflected in the documentation
- [ ] Have you isolated business logic where possible to allow for unit testing?

### Logging and Debugging

- [ ] Are the error messages, if any, informative?
- [ ] Are there enough log events and are they written in a way that allows for easy debugging?
- [ ] "Debugging" code removed
- [ ] Front-end: No erroneous `Console.WriteLines`

### Readability

- [ ] All class, variable, property and method modifiers are provided with the smallest scope possible
- [ ] New files, variables and functions are descriptive/comprehensible and named consistently.
- [ ] There is no dead code (unreachable code)
- [ ] There is no usage of [magic numbers](<https://en.wikipedia.org/wiki/Magic_number_(programming)>)
- [ ] There is no commented out code.
- [ ] In hard-to-understand areas, comments exist and describe rationale or reasons for decisions in code

### Security

- [ ] All personal data inputs are checked (for the correct type, length/size, format, and range).
- [ ] No sensitive information is logged or visible in a stacktrace
- [ ] Are authorization and authentication handled correctly?
- [ ] Is (user) input validated, sanitized, and escaped to prevent security attacks such as cross-site scripting or SQL injection?
- [ ] Is data retrieved from external APIs or libraries checked for security issues?
- [ ] Do API endpoints return appropriate status codes

## Reviewer Guide

- It is important that you understand the purpose of the PR.
- You are encouraged to engage with the requestor if you do not understand any of the proposed code changes/additions/deletions.
- Do you, the reviewer, understand what the code does? Do you think a specific expert, like a security expert or a usability expert, should look over the code before it can be accepted?
- Is a framework, API, library, or service used that should not be used? Are there alternatives you could recommend?
- Are there existing hooks/components/functions in the same code base that could be utilised?
- When reviewing tests, attempt to identify missing edge cases that may be relevant to the proposed implementation
- Ensure you check for:
  - Security
  - Scalability
  - Performance
  - Maintainability

---

> [!NOTE]
> **Medium Risk**
> Touches runtime calculation expression evaluation; the dual-import fallback is low complexity but incorrect resolution would break calculations in some environments.
> 
> **Overview**
> Fixes **form calculation expression parsing** when the package runs as Node ESM against the CommonJS `morph-expressions` package. The default import is no longer used directly as the constructor; it is resolved with `morphExpressions.default ?? morphExpressions` so Node’s interop and bundlers that unwrap `__esModule` both get a usable `ExpressionParser`.
> 
> Type declarations for `morph-expressions` now document that optional `static default` shape.
> 
> Separately, **`@oneblink/release-cli`** is upgraded to `^5.0.1`, and the `release` / `update-dependents` npm scripts pick up the new `--force-update-dependency` (and release adds `--update-dependents`) flags; `package-lock.json` reflects the dependency tree change (including removal of `patch-package` from the release-cli subtree).
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit aff2aab15d49533856728d00d514ed61bae70eea. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: src/calculationService.ts [hunk 1]
```diff
@@ -1,4 +1,9 @@
-import ExpressionParser from 'morph-expressions'
+import morphExpressions from 'morph-expressions'
+// morph-expressions is CommonJS and assigns `exports.default`, which Node's ESM
+// interop exposes as the whole `module.exports` rather than the constructor.
+const ExpressionParser =
+  (morphExpressions as unknown as { default?: typeof morphExpressions })
+    .default ?? morphExpressions
```
Comment (mymattcarroll) [inline src/calculationService.ts]: Is our type def incorrect (cursor wrote it for me last week)? Could it be updated so that we don't need to cast to `unknown`?

File: package-lock.json [hunk 2]
```diff
@@ -4833,9 +4806,6 @@
         "x64"
       ],
       "dev": true,
-      "libc": [
-        "musl"
-      ],
```
Comment (mymattcarroll) [inline package-lock.json:4838]: Do you have latest npm installed? I am on 11.19.0

File: src/calculationService.ts (hunk 1 above)
Comment (divporter) [reply src/calculationService.ts]: 🤷 a cursor job.

File: package-lock.json (hunk 2 above)
Comment (divporter) [reply package-lock.json:4838]: No, but I can be.

File: package.json [hunk 3]
```diff
@@ -14,7 +14,7 @@
     "@eslint/js": "^9.39.2",
     "@microsoft/eslint-plugin-sdl": "^1.1.0",
     "@microsoft/microsoft-graph-types": "^2.43.1",
-    "@oneblink/release-cli": "^4.0.0-beta.2",
+    "@oneblink/release-cli": "^5.0.0",
```
Comment (mymattcarroll) [inline package.json]:
```suggestion
    "@oneblink/release-cli": "^5.0.1",
```

Comment (divporter) [reply package.json]: whoops, thought I already did
